File: tests/algorithms/transpiler/FS/maths/sum_of_geometric_progression.fs
// Generated 2025-08-17 12:28 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec _str v =
    match box v with
    | :? float as f -> sprintf "%.10g" f
    | _ ->
        let s = sprintf "%A" v
        s.Replace("[|", "[")
         .Replace("|]", "]")
         .Replace("; ", " ")
         .Replace(";", "")
         .Replace("\"", "")
let rec pow_float (``base``: float) (exp: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable ``base`` = ``base``
    let mutable exp = exp
    try
        let mutable result: float = 1.0
        let mutable exponent: int = exp
        if exponent < 0 then
            exponent <- -exponent
            let mutable i: int = 0
            while i < exponent do
                result <- result * ``base``
                i <- i + 1
            __ret <- 1.0 / result
            raise Return
        let mutable i: int = 0
        while i < exponent do
            result <- result * ``base``
            i <- i + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
and sum_of_geometric_progression (first_term: int) (common_ratio: int) (num_of_terms: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable first_term = first_term
    let mutable common_ratio = common_ratio
    let mutable num_of_terms = num_of_terms
    try
        if common_ratio = 1 then
            __ret <- float ((int64 num_of_terms) * (int64 first_term))
            raise Return
        let a: float = float first_term
        let r: float = float common_ratio
        __ret <- (a / (1.0 - r)) * (1.0 - (pow_float (r) (num_of_terms)))
        raise Return
        __ret
    with
        | Return -> __ret
and test_sum () =
    let mutable __ret : obj = Unchecked.defaultof<obj>
    try
        if (sum_of_geometric_progression (1) (2) (10)) <> 1023.0 then
            ignore (failwith ("example1 failed"))
        if (sum_of_geometric_progression (1) (10) (5)) <> 11111.0 then
            ignore (failwith ("example2 failed"))
        if (sum_of_geometric_progression (-1) (2) (10)) <> (-1023.0) then
            ignore (failwith ("example3 failed"))
        __ret
    with
        | Return -> __ret
and main () =
    let mutable __ret : obj = Unchecked.defaultof<obj>
    try
        let __bench_start = _now()
        let __mem_start = System.GC.GetTotalMemory(true)
        ignore (test_sum())
        ignore (printfn "%s" (_str (sum_of_geometric_progression (1) (2) (10))))
        let __bench_end = _now()
        let __mem_end = System.GC.GetTotalMemory(true)
        printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

        __ret
    with
        | Return -> __ret
ignore (main())

File: tests/algorithms/x/PHP/maths/sum_of_geometric_progression.php
<?php
error_reporting(E_ALL & ~E_DEPRECATED);
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _panic($msg) {
    fwrite(STDERR, strval($msg));
    exit(1);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function pow_float($base, $exp) {
  $result = 1.0;
  $exponent = $exp;
  if ($exponent < 0) {
  $exponent = -$exponent;
  $i = 0;
  while ($i < $exponent) {
  $result = $result * $base;
  $i = $i + 1;
};
  return 1.0 / $result;
}
  $i = 0;
  while ($i < $exponent) {
  $result = $result * $base;
  $i = $i + 1;
};
  return $result;
};
  function sum_of_geometric_progression($first_term, $common_ratio, $num_of_terms) {
  if ($common_ratio == 1) {
  return floatval(($num_of_terms * $first_term));
}
  $a = floatval($first_term);
  $r = floatval($common_ratio);
  return ($a / (1.0 - $r)) * (1.0 - pow_float($r, $num_of_terms));
};
  function test_sum() {
  if (sum_of_geometric_progression(1, 2, 10) != 1023.0) {
  _panic('example1 failed');
}
  if (sum_of_geometric_progression(1, 10, 5) != 11111.0) {
  _panic('example2 failed');
}
  if (sum_of_geometric_progression(-1, 2, 10) != (-1023.0)) {
  _panic('example3 failed');
}
};
  function main() {
  test_sum();
  echo rtrim(json_encode(sum_of_geometric_progression(1, 2, 10), 1344)), PHP_EOL;
};
  main();
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;
